numba/tests/test_npdatetime.py

```diff
@@ -903,6 +944,153 @@ def test_add_td_no_match(self):
+                DT("2004-12-07", "ns"),
+                DT("NaT", "ns"),
+            ]),
+            np.array([
+                DT("NaT", "ns"),
+                DT("2011-01-01", "ns"),
+                DT("1971-02-02", "ns"),
+                DT("1900-01-01", "ns"),
+                DT("2021-03-03", "ns"),
+                DT("2004-12-07", "ns"),
+            ]),
+            np.array([
+                DT("NaT", "ns"),
+            ]),
+            np.array([
+                DT("NaT", "ns"),
+                DT("NaT", "ns"),
+                DT("NaT", "ns"),
+            ]),
+            np.array([
+                TD(1, "ns"),
+                TD(2, "ns"),
+                TD(3, "ns"),
+                TD(4, "ns"),
+            ]),
+            np.array([
+                TD(1, "D"),
+                TD(2, "D"),
+                TD(3, "D"),
+                TD(4, "D"),
+            ]),
+            np.array([
+                TD("NaT", "ns"),
+                TD(1, "ns"),
+                TD(2, "ns"),
+                TD(3, "ns"),
+                TD(4, "ns"),
+            ]),
+            np.array([
+                TD(1, "ns"),
+                TD(2, "ns"),
+                TD(3, "ns"),
+                TD(4, "ns"),
+                TD("NaT", "ns"),
+            ]),
+            np.array([
+                TD("NaT", "ns"),
+            ]),
+            np.array([
+                TD("NaT", "ns"),
+                TD("NaT", "ns"),
+                TD("NaT", "ns"),
+            ]),
+        ]
+
+        for arr in test_cases:
+            py_res = py_func(arr)
+            c_res = cfunc(arr)
+            if np.isnat(py_res) or np.isnat(c_res):
+                self.assertTrue(np.isnat(py_res))
+                self.assertTrue(np.isnat(c_res))
+            else:
+                self.assertEqual(py_res, c_res)
+
+    def test_min_func(self):
+        self._test_min_max(min, False, False)
+
+    def test_np_min_func(self):
+        self._test_min_max(np.min, False, False)
+
+    @skip_parfors_unsupported
+    def test_min_func_parallel(self):
+        self._test_min_max(min, True, False)
+
+    @skip_parfors_unsupported
+    def test_np_min_func_parallel(self):
+        self._test_min_max(np.min, True, False)
+
+    def test_max_func(self):
+        self._test_min_max(max, False, False)
+
+    def test_np_max_func(self):
+        self._test_min_max(np.max, False, False)
+
+    @skip_parfors_unsupported
+    def test_max_func_parallel(self):
+        self._test_min_max(max, True, False)
+
+    @skip_parfors_unsupported
+    def test_np_max_func_parallel(self):
+        self._test_min_max(np.max, True, False)
+
+    def test_min_method(self):
+        self._test_min_max(np.min, False, True)
+
+    @skip_parfors_unsupported
+    def test_min_method_parallel(self):
+        self._test_min_max(np.min, True, True)
+
+    def test_max_method(self):
+        self._test_min_max(np.max, False, True)
+
+    @skip_parfors_unsupported
```

Comment:
I think it might be helpful if you group the parallel tests together. Switching between parallel and regular makes it hard to keep track of which tests are needed.